db/db_iter.cc

```diff
@@ -289,6 +305,24 @@ bool DBIter::SetValueAndColumnsFromMergeResult(const Status& merge_status,
   return true;
 }
 
+bool DBIter::PrepareValue() {
+  assert(valid_);
+
+  if (lazy_blob_index_.empty()) {
+    return true;
```

Comment:
This is in case `PrepareValue` is called repeatedly, right? Shall we return `valid_` here?

Reply:
It also covers the case when the value is not a blob (e.g. BlobDB is actually disabled, or the value size is below `min_blob_size`).

Re: returning `valid_`, `PrepareValue` should only be called if the iterator is valid; this is documented in the API comment and we also have an assertion above.